pkg/scheduler/schedule_one.go

```diff
@@ -164,46 +170,35 @@ func (sched *Scheduler) schedulingCycle(ctx context.Context, state *framework.Cy
 	// assume modifies `assumedPod` by setting NodeName=scheduleResult.SuggestedHost
 	err = sched.assume(assumedPod, scheduleResult.SuggestedHost)
 	if err != nil {
-		metrics.PodScheduleError(fwk.ProfileName(), metrics.SinceInSeconds(start))
 		// This is most probably result of a BUG in retrying logic.
 		// We report an error here so that pod scheduling can be retried.
 		// This relies on the fact that Error will check if the pod has been bound
 		// to a node and if so will not add it back to the unscheduled pods queue
 		// (otherwise this would cause an infinite loop).
-		sched.FailureHandler(ctx, fwk, assumedPodInfo, err, v1.PodReasonSchedulerError, clearNominatedNode)
-		return ScheduleResult{}, nil
+		return ScheduleResult{nominatingInfo: clearNominatedNode, reason: v1.PodReasonSchedulerError},
+			assumedPodInfo,
+			err
 	}
 
 	// Run the Reserve method of reserve plugins.
 	if sts := fwk.RunReservePluginsReserve(ctx, state, assumedPod, scheduleResult.SuggestedHost); !sts.IsSuccess() {
-		metrics.PodScheduleError(fwk.ProfileName(), metrics.SinceInSeconds(start))
-		// trigger un-reserve to clean up state associated with the reserved Pod
-		fwk.RunReservePluginsUnreserve(ctx, state, assumedPod, scheduleResult.SuggestedHost)
-		if forgetErr := sched.Cache.ForgetPod(assumedPod); forgetErr != nil {
-			klog.ErrorS(forgetErr, "Scheduler cache ForgetPod failed")
-		}
-		sched.FailureHandler(ctx, fwk, assumedPodInfo, sts.AsError(), v1.PodReasonSchedulerError, clearNominatedNode)
-		return ScheduleResult{}, nil
+		return ScheduleResult{nominatingInfo: clearNominatedNode, reason: v1.PodReasonSchedulerError, unreserve: true},
```

Comment:
I prefer to leave the calls to UnReserve and ForgetPod here, instead.

Reply:
There is only 2 instances where we call Unreserve and Forget during scheduling cycle. Here and after Permit.

It sounds pretty clean to me to just keep the calls here.

bindingCycle handles it differently, but that's not relevant here. In other words, I think we can keep `handleBindingCycleError`

Reply:
done